src/include/sof/audio/component.h

```diff
@@ -15,15 +15,17 @@
 
 #ifndef __SOF_AUDIO_COMPONENT_H__
 #define __SOF_AUDIO_COMPONENT_H__
-
+#ifndef MODULE_PRIVAT
```

Comment:
we are using __SOF_MODULE_SERVICE_BUILD__ for loadable module, please check loadable smart_amp